tools/autograd/gen_variable_type.py

```diff
@@ -767,10 +787,41 @@ def declare_returned_variables():
         names = [ret['type'] + ' ' + ret['name'] + ';' for ret in declaration['returns']]
         return '\n'.join(names)
 
-    def wrap_output(call):
-        # Returns `wrapped_call` which is a drop-in replacement for `call`
+    def emit_view_lambda():
+        input_base = 'input_base'
+        replay_view_func = ''
+        updated_unpacked_args = []
+        combined = nested_dict(env, declaration)
+        for arg in combined['unpacked_args']:
+            if arg == 'self_':
+                updated_unpacked_args.append(input_base)
+            elif combined['unpacked_args_simple_type'][arg] == 'IntArrayRef':
```

Comment:
A short comment here saying "it's not safe to close over IntArrayRef by value, since this is a reference type, so materialize a vector to close over by value" would go a long way to explaining what is going on here

Reply:
It is kind of unfortunate that you can't reuse `save_variables` here, the logic is pretty similar. 

Would it be possible to post ALL view functions that got lambdas generated this way? I'm now curious if you can ever close over a Tensor here: if you do so, there is risk of inducing a reference cycle (which is why we have the SavedVariable business)

Reply:
Yea all code generated here are in https://gist.github.com/ailzhang/5129517bd630579ef69536afe33255cb. 
We don't have to worry about `view_as` as these functions are not dispatched as `use_derived`. In the end it'll dispatch to `view` and we only generate this function for non-composite ops like `view`. 
But you're definitely right that I should make it clear that it works now since all arguments we close over by value are working. And if we ever want to add use_derived op with Tensor type, it should send out proper error messages.

Reply:
And I think it might be okay to not merge this logic to `save_variables` since this materialization logic is pretty unique and here we only care about a very limit subset of types. `save_variables` is already complicated so I feel keeping it separate for now should be fine.

Reply:
Yeah, I'm fine with not merging.